From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ysl3000 <yannicklamprecht@live.de>
Date: Sat, 22 Aug 2020 16:23:52 +0200
Subject: [PATCH] fixed command data missmatch


diff --git a/src/main/java/net/minecraft/core/BlockPos.java b/src/main/java/net/minecraft/core/BlockPos.java
index 24e97d4e2c5777578bb7ccc57071b83ac66aa420..2c96aacc01e98cb9a2f4e9e81a8529aa82f223b6 100644
--- a/src/main/java/net/minecraft/core/BlockPos.java
+++ b/src/main/java/net/minecraft/core/BlockPos.java
@@ -118,6 +118,9 @@ public class BlockPos extends Vec3i {
         return this.offset(-baseblockposition.getX(), -baseblockposition.getY(), -baseblockposition.getZ());
     }
 
+    public BlockPos up() { // Toothpick
+        return above();
+    }
     @Override
     public BlockPos above() {
         return new BlockPos(this.getX(), this.getY() + 1, this.getZ()); // Paper - Optimize BlockPosition
